module/os/linux/zfs/zfs_vfsops.c

```diff
@@ -2275,6 +2278,7 @@ zfs_resume_fs(zfsvfs_t *zfsvfs, dsl_dataset_t *ds)
 	    zp = list_next(&zfsvfs->z_all_znodes, zp)) {
 		err2 = zfs_rezget(zp);
 		if (err2) {
+			spl_d_drop_aliases(ZTOI(zp));
```

Comment:
Additionally, there's going to need to be some mechanism to drop any stale negative dentries.  On a quick reading of `shrink_dcache_sb()` it wasn't clear to me if we could count on it destroying these.

Reply:
At least according to my testing, `shrink_dcache_sb()` does take care of the negative dentries, but I missed the positive open file dentries last time...